File: JavaLox/src/com/craftinginterpreters/lox/RuntimeError.java
package com.craftinginterpreters.lox;

/**
 *
 * @author brusc
 */
class RuntimeError extends RuntimeException {
   /**
    * 
    */
   final Token token;
   
   /**
    * 
    * @param token
    * @param message 
    */
   RuntimeError(Token token, String message) {
      super(message);
      this.token = token;
   }

   /**
    * 
    * @param message
    * @param cause
    * @param enableSuppression
    * @param writableStackTrace 
    */
   RuntimeError(String message, Object cause, boolean enableSuppression, boolean writableStackTrace) {
      super(message, (Throwable)cause, enableSuppression, writableStackTrace);
      token = null;
   }
}

File: CsLox/com/craftinginterpreters/lox/RuntimeError.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace com.craftinginterpreters.lox
{
    /// <summary>
    /// 
    /// </summary>
    internal class RuntimeError : Exception
    {
        /// <summary>
        /// 
        /// </summary>
        internal readonly Token token;

        /// <summary>
        /// 
        /// </summary>
        /// <param name="token"></param>
        /// <param name="message"></param>
        internal RuntimeError(Token token, String message): base(message)
        {
            this.token = token;    
        }

        /// <summary>
        /// 
        /// </summary>
        /// <param name="message"></param>
        /// <param name="cause"></param>
        /// <param name="enableSuppression"></param>
        /// <param name="writableStackTrace"></param>
        internal RuntimeError(String message, Object cause, bool enableSuppression, bool writableStackTrace) : base(message)
        {
            token = null;
        }
    }
}
